File: testdata/bench/kotlin/BenchmarkTest00075.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.io.File

// Safe: allowlist of filenames
suspend fun handler00075(call: ApplicationCall) {
    val doc = call.parameters["doc"]
    val allowed = setOf("readme.txt", "license.txt", "changelog.txt")
    if (doc !in allowed) {
        call.respondText("Not found")
        return
    }
    val content = File("/docs/$doc").readText()
    call.respondText(content)
}

File: testdata/bench/swift/BenchmarkTest00075.swift
import Vapor
import Foundation

func handler075(_ req: Request) throws -> Response {
    let file = try req.parameters.require("file")
    let allowed = ["readme.txt", "license.txt", "changelog.txt"]
    guard allowed.contains(file) else { throw Abort(.notFound) }
    let data = FileManager.default.contents(atPath: "/docs/" + file)
    return Response(status: .ok, body: .init(data: data ?? Data()))
}
